src/bio/bio_config.c

```diff
@@ -699,3 +713,96 @@ bio_set_hotplug_filter(const char *nvme_conf) {
 
 	return rc;
 }
+
+static int
+get_accel_props(const char *nvme_conf)
+{
+	struct json_config_ctx	*ctx;
+	struct spdk_json_val	*daos_data;
+	struct config_entry	 cfg = {};
+	int			 rc = 0;
+
+	D_ASSERT(nvme_conf != NULL);
+
+	D_ALLOC_PTR(ctx);
+	if (ctx == NULL)
+		return -DER_NOMEM;
+
+	rc = read_config(nvme_conf, ctx);
+	if (rc < 0) {
+		D_ERROR("No config file\n");
+		goto out;
+	}
+
+	/* Capture daos object */
+	rc = spdk_json_find(ctx->values, "daos_data", NULL, &daos_data,
+			    SPDK_JSON_VAL_OBJECT_BEGIN);
+	if (rc < 0) {
+		D_ERROR("Failed to find 'daos_data' key: %s\n", strerror(-rc));
+		rc = -DER_INVAL;
+		goto out;
+	}
+
+	/* Capture config array in ctx */
+	rc = spdk_json_decode_object(daos_data, daos_data_decoders,
+				     SPDK_COUNTOF(daos_data_decoders), ctx);
+	if (rc < 0) {
+		D_ERROR("Failed to parse 'daos_data' entry: %s\n", strerror(-rc));
+		rc = -DER_INVAL;
+		goto out;
+	}
+
+	/* Get 'config' array first configuration entry */
+	ctx->config_it = spdk_json_array_first(ctx->config);
+	if (ctx->config_it == NULL) {
+		D_DEBUG(DB_MGMT, "Empty 'daos_data' section\n");
+		goto out; /* non-fatal */
+	}
+
+	while (ctx->config_it != NULL) {
+		rc = spdk_json_decode_object(ctx->config_it, config_entry_decoders,
+					     SPDK_COUNTOF(config_entry_decoders), &cfg);
+		if (rc < 0) {
+			D_ERROR("Failed to decode 'config' entry: %s\n", strerror(-rc));
+			rc = -DER_INVAL;
+			goto out;
+		}
+
+		if (strcmp(cfg.method, NVME_CONF_SET_ACCEL_PROPS) == 0)
+			break;
+
+		/* Move on to next subsystem config */
+		ctx->config_it = spdk_json_next(ctx->config_it);
+	}
+
+	if (ctx->config_it == NULL) {
+		D_DEBUG(DB_MGMT, "No '%s' entry\n", NVME_CONF_SET_ACCEL_PROPS);
+		goto out; /* non-fatal */
+	}
+
+	rc = spdk_json_decode_object(cfg.params, accel_props_decoders,
+				     SPDK_COUNTOF(accel_props_decoders),
+				     &accel_props);
+	if (rc < 0) {
+		D_ERROR("Failed to decode '%s' entry (rc: %d)\n", NVME_CONF_SET_ACCEL_PROPS, rc);
+		rc = -DER_INVAL;
+		goto out;
+	}
+
+	D_INFO("'%s' read from config, setting: %s, capabilities: move=%s,crc=%s\n",
+	       NVME_CONF_SET_ACCEL_PROPS, accel_props.engine,
+	       CHK_FLAG(accel_props.opt_mask, NVME_ACCEL_FLAG_MOVE) ? "true" : "false",
+	       CHK_FLAG(accel_props.opt_mask, NVME_ACCEL_FLAG_CRC) ? "true" : "false");
```

Comment:
This is being imported from control.h? I'm starting to question the value of having that single file as a catch-all that includes all of our C constants. We should probably have a daos_srv/bio_types.h for the domain-specific constants/types that need to be shared, if we don't want everything in bio.h.